### PR TITLE
router: add missing unit tests for x-envoy-attempt-count on upstream requests

### LINKED ISSUE #10366: router: add unit tests for x-envoy-attempt-count on the request path
I noticed while working on https://github.com/envoyproxy/envoy/pull/10325 that we did not have unit tests for x-envoy-attempt-count behavior on the request path similar to the ones added for the response path on #10325, specially for overwritten behavior. There are a couple integration tests that ensure basic functionality, and config tests.

### PR DESCRIPTION
**Pull request by junr03**
Description: adding missing unit test coverage.
Risk Level: low
Testing: added new tests

Fixes #10366

### CODE REVIEW

File: test/common/router/router_test.cc [hunk 1]
```diff
@@ -868,6 +872,121 @@ TEST_F(RouterTest, EnvoyUpstreamServiceTime) {
   EXPECT_TRUE(verifyHostUpstreamStats(1, 0));
 }
 
+// Validate that x-envoy-attempt-count is added to request headers when option is true.
```
Comment (alyssawilk) [inline test/common/router/router_test.cc]: when option -> when the option ?

File: test/common/router/router_test.cc [hunk 2]
```diff
@@ -868,6 +872,121 @@ TEST_F(RouterTest, EnvoyUpstreamServiceTime) {
...
+TEST_F(RouterTest, EnvoyAttemptCountInRequest) {
+  setIncludeAttemptCount(true);
```
Comment (alyssawilk) [inline test/common/router/router_test.cc]: optional: do you want to name this to Request even though it doesn't reflect the config option until v4?

File: test/common/router/router_test.cc [hunk 3]
```diff
@@ -868,6 +872,121 @@ TEST_F(RouterTest, EnvoyUpstreamServiceTime) {
...
+
+  EXPECT_CALL(cm_.conn_pool_, newStream(_, _)).WillOnce(Return(&cancellable_));
+  expectResponseTimerCreate();
+
+  Http::TestRequestHeaderMapImpl headers;
+  HttpTestUtility::addDefaultHeaders(headers);
+  router_.decodeHeaders(headers, true);
```
Comment (alyssawilk) [inline test/common/router/router_test.cc]: Also optional but what do you think of setting up some utility functions to reduce cut and paste code here?

File: test/common/router/router_test.cc (hunk 2 above)
Comment (junr03) [reply test/common/router/router_test.cc]: I wanted to keep the name change in #10325. So I will update this or that other PR depending on merge order.

**Comment by junr03** [comment]
@alyssawilk updated!

**Pull request merged**